libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Core.hs

```diff
@@ -317,22 +319,38 @@ conformsToImpl ::
   , FixupSpecRep (SpecRep (ExecState fn rule era))
   ) =>
   Property
-conformsToImpl =
-  let genCtx = genExecContext @fn @rule @era
-   in forAllShow genCtx showExpr $ \ctx ->
-        let envSpec = simplifySpec $ environmentSpec @fn @rule @era ctx
-         in forAllShrinkShow (CV2.genFromSpec envSpec) (shrinkWithSpec envSpec) showExpr $ \env ->
-              let stSpec = simplifySpec $ stateSpec @fn @rule @era ctx env
-               in forAllShrinkShow (CV2.genFromSpec stSpec) (shrinkWithSpec stSpec) showExpr $ \st ->
-                    let sigSpec = simplifySpec $ signalSpec @fn @rule @era ctx env st
-                     in forAllShrinkShow (CV2.genFromSpec sigSpec) (shrinkWithSpec sigSpec) showExpr $ \sig ->
-                          counterexample (extraInfo @fn @rule @era ctx (inject env) (inject st) (inject sig)) $ do
-                            _ <- impAnn @_ @era "Deep evaluating env" $ evaluateDeep env
-                            _ <- impAnn "Deep evaluating st" $ evaluateDeep st
-                            _ <- impAnn "Deep evaluating sig" $ evaluateDeep sig
```

Comment:
Looks like you lost forcing to NF. Was that intentional?

Reply:
Good catch